# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsGfxScrollFrame.cpp
# Commit: ea3a73c7788e
# Full Hash: ea3a73c7788e2b9752d7f00976c8119183e3da18
# Author: Matt Woodrow <mwoodrow@mozilla.com>
# Date: 2019-05-07 09:44:39
# Description:
#   Bug 1548483 - Don't assert that we have an invalidated frame when we encouter a new displayport, since it can have changed on an earlier paint if we didn't have display items. r=tnikkel
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D29975
# ==============================================================================

diff -r ef8c4674d0b0 -r ea3a73c7788e layout/generic/nsGfxScrollFrame.cpp
--- a/layout/generic/nsGfxScrollFrame.cpp	Mon May 06 20:56:16 2019 +0000
+++ b/layout/generic/nsGfxScrollFrame.cpp	Mon May 06 09:13:00 2019 +0000
@@ -3931,10 +3931,6 @@
   // date if we just introduced a new animated geometry root.
   if (oldWillBuildScrollableLayer != mWillBuildScrollableLayer) {
     aBuilder->RecomputeCurrentAnimatedGeometryRoot();
-    MOZ_DIAGNOSTIC_ASSERT(!aBuilder->IsPartialUpdate() ||
-                              aBuilder->InInvalidSubtree() ||
-                              mOuter->IsFrameModified(),
-                          "Displayport changed without an invalidation");
   }
 
   mIsScrollableLayerInRootContainer =
